packages/api/src/tasks/checkAlerts.ts

```diff
@@ -262,18 +262,26 @@ export const handleSendGenericWebhook = async (
     'Content-Type': 'application/json', // default, will be overwritten if user has set otherwise
     ...(webhook.headers?.toJSON() ?? {}),
   };
-
   // BODY
-
   let body = '';
-  if (webhook.body) {
-    const handlebars = Handlebars.create();
-    body = handlebars.compile(webhook.body, {
-      noEscape: true,
-    })({
-      body: escapeJsonString(message.body),
-      link: escapeJsonString(message.hdxLink),
-      title: escapeJsonString(message.title),
+  try {
+    if (webhook.body) {
+      const handlebars = Handlebars.create();
+      body = handlebars.compile(webhook.body, {
+        noEscape: true,
+      })({
+        body: escapeJsonString(message.body),
+        link: escapeJsonString(message.hdxLink),
+        title: escapeJsonString(message.title),
+      });
+    }
+  } catch (e) {
+    logger.error({
+      message: 'Failed to compile generic webhook body',
+      error: serializeError(e),
+    });
+    body = JSON.stringify({
+      text: `${escapeJsonString(message.title)} | ${escapeJsonString(message.body)} | ${escapeJsonString(message.hdxLink)}`,
```

Comment:
Any reason we want to set the default body here? I think aborting is more appropriate given the template is invalid

Reply:
The idea is we still send alert with info instead of silence it, so customer can have better idea on what's going on(not confuse on why they did not receive alert).

Reply:
Hrmm... for a generic webhook that _might_ be inappropriate given they might expect one payload format and getting another would be weird and can lead to undefined behavior on the user's end. (Imagine they have an app that receives this input and does an action on it, an invalid payload may lead to bad behavior).

Ideally we notify the user somehow through another channel instead. The best we can do here in OSS is probably to drop an error log.

Reply:
Ah get it, I can revert this part. Also I will abort alert if error happen at try catch. Sounds good? @wrn14897 @MikeShi42

Reply:
That sounds good to me, we can revisit improving this flow later.

Reply:
Sounds good to me